src/react/HomePage.tsx

```diff
@@ -241,94 +304,104 @@ const TriggersCard = ({ triggerList, onInstallTriggerClick }) => (
-      </td>
-    );
-  } else {
-    messageCell = (
-      <td className="color-red">
-        <i className="fas fa-times-circle mr-1" />
-        This service appears offline
-      </td>
-    );
-  }
+export default ({ astarte }: Props): React.ReactElement => {
+  const devicesStats = useFetch(astarte.getDevicesStats);
+  const interfaces = useFetch(astarte.getInterfaceNames);
+  const triggers = useFetch(astarte.getTriggerNames);
+  const appEngineHealth = useFetch(astarte.getAppengineHealth);
+  const realmManagementHealth = useFetch(astarte.getRealmManagementHealth);
+  const pairingHealth = useFetch(astarte.getPairingHealth);
+  const flowHealth = astarte.features.flow ? useFetch(astarte.getFlowHealth) : null;
+  const navigate = useNavigate();
 
-  return (
-    <tr>
-      <td>{service}</td>
-      {messageCell}
-    </tr>
-  );
-};
+  const connectedDevicesProvider = useMemo(() => getConnectedDevices(astarte), [astarte]);
 
-const InterfaceList = ({ interfaces, onInterfaceClick, maxShownInterfaces }) => {
-  const shownInterfaces = interfaces.slice(0, maxShownInterfaces);
-  const remainingInterfaces = interfaces.length - maxShownInterfaces;
+  const refreshData = () => {
+    devicesStats.refresh();
+    interfaces.refresh();
+    triggers.refresh();
+    appEngineHealth.refresh();
+    realmManagementHealth.refresh();
+    pairingHealth.refresh();
+    if (astarte.features.flow && flowHealth) {
+      flowHealth.refresh();
+    }
+  };
 
-  return (
-    <ul className="list-unstyled">
-      {shownInterfaces.map((interfaceName) => (
-        <li key={interfaceName} className="my-1">
-          <a
-            href="/interfaces"
-            onClick={(e) => {
-              onInterfaceClick(e, interfaceName);
-            }}
-          >
-            <i className="fas fa-stream mr-1" />
-            {interfaceName}
-          </a>
-        </li>
-      ))}
-      {remainingInterfaces > 0 && (
-        <li>
-          <Link to="/interfaces">
-            {`${remainingInterfaces} more installed ${
-              remainingInterfaces > 1 ? 'interfaces' : 'interface'
-            }…`}
-          </Link>
-        </li>
-      )}
-    </ul>
-  );
-};
+  useEffect(() => {
+    const refreshTimer = setInterval(() => {
+      refreshData();
+    }, 30000);
+    astarte.addListener('credentialsChange', refreshData);
 
-const TriggerList = ({ triggers, maxShownTriggers }) => {
-  const shownTriggers = triggers.slice(0, maxShownTriggers);
-  const remainingTriggers = triggers.length - maxShownTriggers;
+    return () => {
+      clearTimeout(refreshTimer);
+      astarte.removeListener('credentialsChange', refreshData);
+    };
+  }, [astarte]);
+
+  const redirectToLastInterface = useCallback((e, interfaceName) => {
+    e.preventDefault();
+    astarte.getInterfaceMajors(interfaceName).then((interfaceMajors) => {
+      const latestMajor = Math.max(...interfaceMajors);
+      navigate(`/interfaces/${interfaceName}/${latestMajor}`);
```

Comment:
`/${latestMajor}/edit`?
I guess the problem depends on what gets merged first, this or the forward-port from 1.0

Reply:
Let's just wait a bit and merge the forward-port first.